client/src/components/MeetingExperience.tsx

```diff
@@ -75,12 +89,26 @@ export const MeetingExperience = (props: MeetingExperienceProps): JSX.Element =>
 
     _createAdapters();
   }, [credential, displayName, endpointUrl, locator, userId, onDisplayError]);
-
   if (callWithChatAdapter) {
     const logo = logoUrl ? <img style={meetingExperienceLogoStyles} src={logoUrl} /> : <></>;
     const locale = COMPOSITE_LOCALE_EN_US;
     const formFactorValue = new MobileDetect(window.navigator.userAgent).mobile() ? 'mobile' : 'desktop';
 
+    if (renderPostCall) {
+      return (
+        <Survey
+          postCall={postCall}
+          onRejoinCall={() => {
+            callWithChatAdapter.joinCall();
+            callWithChatAdapter.onStateChange((state: CallWithChatAdapterState) => {
```

Comment:
Feels like the order of methods should be changed? First, we subscribe to events that happen when we `joinCall` and then we initiate the actual `joinCall`.

Reply:
Tested this change and it works both ways. But yes, reads better to subscribe first and trigger after.

Reply:
have made this change.

Reply:
From UX review it is seen that when we end call on "Joining Call" screen (lobby hasnt fully loaded, the survey page gets skipped. This behaviour is avoided when joinCall is called first and subscribing to onStateChanged happens right after. So I think it would be better to revert.

Reply:
Ughh.. actually that doesnt fix it. Will have to do some more digging to see what a potential fix might be